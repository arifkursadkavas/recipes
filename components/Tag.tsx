import * as React from 'react';

const Tag: React.FC<TagProps> = (props) => {
  const { text } = props;
  return (
    <div className="tagContainer">
      <span className="tagText">{text}</span>
    </div>
  );
};

interface TagProps {
  text: string;
}

export default Tag;
