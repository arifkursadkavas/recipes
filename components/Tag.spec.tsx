import * as React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Tag from './Tag';

describe('Components', () => {
  describe('Tag', () => {
    test('snapshot renders', () => {
      const component = renderer.create(<Tag text="Tag Text" />);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe('Components', () => {
  describe('Tag', () => {
    it('should render Tag text without throwing an error', function () {
      const wrap = mount(<Tag text="Tag Text" />);
      expect(wrap.find('.tagText').text()).toBe('Tag Text');
    });
  });
});
