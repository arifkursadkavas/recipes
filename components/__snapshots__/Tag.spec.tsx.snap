// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Tag snapshot renders 1`] = `
<div
  className="tagContainer"
>
  <span
    className="tagText"
  >
    Tag Text
  </span>
</div>
`;
